### PR TITLE
DSL syntax refactor

### LINKED ISSUE #56: As a framework user I would higly appreciate a handy DSL to perform these operations
**Initial estimate of effort:** 40

#### Sprint tasks
- [x] #57
- [x] #58
- [x] #79 
- [x] #80
- [x] #85
- [x] #108
- [x] #109

### LINKED ISSUE #85: Complete DSL for Systems

### LINKED ISSUE #108: Use new System builder in DSL test

### PR DESCRIPTION
**Pull request by vitlinda**
This PR introduces a major refactor of the DSL, a removeSystem method (both in `World's` trait and in the DSL) and adds the DSL syntax for the excluding `View`.
Closes #85 
Closes #108

### CODE REVIEW

**Review by ndido98 — APPROVED** [review]

File: core/src/main/scala/dev/atedeg/ecscala/dsl/Syntax.scala [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    override def from(world: World): Unit = world.removeSystem(system)
+  }
+
+  /**
+   * This case class enables the following syntax:
+   * {{{
+   *   clearAll from world
+   * }}}
+   */
+  case class ClearAllFromWorld(clearWord: ClearWord) extends From[World, Unit] {
```
Comment (giacomocavalieri) [inline core/src/main/scala/dev/atedeg/ecscala/dsl/Syntax.scala:44]: Why does it take a `ClearWord` as a parameter if it does nothing with it? I would remove the `ClearWord` altogether

File: core/src/main/scala/dev/atedeg/ecscala/dsl/Syntax.scala [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  case class EntityWord()
+
+  /**
+   * This case class enables the following syntax:
+   *
+   * {{{
+   * clearAll from world
+   * }}}
+   */
+  private[dsl] case class ClearWord()
```
Comment (giacomocavalieri) [inline core/src/main/scala/dev/atedeg/ecscala/dsl/Syntax.scala:111]: I would remove this as it serves no purpose, read my comment above

File: core/src/test/scala/dev/atedeg/ecscala/WorldTest.scala [hunk 3]
```diff
@@ -102,5 +102,25 @@ class WorldTest extends AnyWordSpec with Matchers {
...
+        entity.addComponent(Position(1, 1))
+        val system = new System[Position &: CNil] {
+          override def update(
+              entity: Entity,
+              components: Position &: CNil,
+          )(deltaTime: DeltaTime, world: World, view: View[Position &: CNil]): Deletable[Position &: CNil] = {
+            val Position(x, y) &: CNil = components
+            Position(x + 1, y + 1) &: CNil
+          }
+        }
```
Comment (giacomocavalieri) [inline core/src/test/scala/dev/atedeg/ecscala/WorldTest.scala:117]: Why didn't you use the `SystemBuilder`

File: core/src/test/scala/dev/atedeg/ecscala/dsl/ECScalaDSLTest.scala [hunk 4]
```diff
@@ -118,11 +119,53 @@ class ECScalaDSLTest extends AnyWordSpec with Matchers with ECScalaDSL {
...
+
+      val mySystem = new System[Position &: CNil] {
+        override def update(
+            entity: Entity,
+            components: Position &: CNil,
+        )(deltaTime: DeltaTime, world: World, view: View[Position &: CNil]) = {
+          val Position(x, y) &: CNil = components
+          Position(x + 3, y + 3)
+        }
+      }
```
Comment (giacomocavalieri) [inline core/src/test/scala/dev/atedeg/ecscala/dsl/ECScalaDSLTest.scala:131]: Why didn't you use the `SystemBuilder`

File: core/src/test/scala/dev/atedeg/ecscala/dsl/ECScalaDSLTest.scala [hunk 5]
```diff
@@ -118,11 +119,53 @@ class ECScalaDSLTest extends AnyWordSpec with Matchers with ECScalaDSL {
...
+      val entity1 = world hasAn entity withComponent Position(1, 1)
+      val aSystem = new System[Position &: CNil] {
+        override def update(
+            entity: Entity,
+            components: Position &: CNil,
+        )(deltaTime: DeltaTime, world: World, view: View[Position &: CNil]): Deletable[Position &: CNil] = {
+          val Position(x, y) &: CNil = components
+          Position(x + 1, y + 1) &: CNil
+        }
+      }
```
Comment (giacomocavalieri) [inline core/src/test/scala/dev/atedeg/ecscala/dsl/ECScalaDSLTest.scala:157]: Why didn't you use the `SystemBuilder`?

File: core/src/test/scala/dev/atedeg/ecscala/dsl/ECScalaDSLTest.scala [hunk 6]
```diff
@@ -133,6 +176,20 @@ class ECScalaDSLTest extends AnyWordSpec with Matchers with ECScalaDSL {
     }
   }
 
+  "geView[Position &: CNil] excluding[Velocity &: CNil] from world" should {
```
Comment (giacomocavalieri) [inline core/src/test/scala/dev/atedeg/ecscala/dsl/ECScalaDSLTest.scala:179]:
```suggestion
  "geView[Position &: CNil].excluding[Velocity &: CNil] from world" should {
```

**Review by giacomocavalieri — CHANGES_REQUESTED** [review]

File: core/src/main/scala/dev/atedeg/ecscala/dsl/Syntax.scala [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+
+import dev.atedeg.ecscala.{ CList, Component, Entity, System, View, World }
+import dev.atedeg.ecscala.util.types.{ CListTag, ComponentTag }
+import dev.atedeg.ecscala.util.types.given
+import dev.atedeg.ecscala.dsl.Words.*
+
+trait Syntax {
+
+  /**
+   * This trait enables the use of the word "from" in the dsl
```
Comment (nicolasfara) [inline core/src/main/scala/dev/atedeg/ecscala/dsl/Syntax.scala:11]: Missing a `.` at the end of the sentence

**Review by nicolasfara — CHANGES_REQUESTED** [review]

**Review by nicolasfara — APPROVED** [review]

**Review by giacomocavalieri — APPROVED** [review]

**Pull request merged**